### ISSUE TITLE
Captions don't work with AMD hardware encoder on Windows

### ISSUE THREAD

**Issue by MuzikxMonkey** [ratwithacompiler/OBS-captions-plugin#8]
The caption dock works, it captures my text perfectly, but when i check this on stream, I dont have a CC option. I have checked from an android phone, an iPhone, and a PC. I tried the latest .11 version, i even rolled back thru to the .10 and the initial releases. None of them work. Is there something I may be missing along with this?? Thank you for any reply.

**Comment by Swindyyy** [comment]
You're not alone, mine does not work either using the latest release. I'm tempted to go back a few just to see if it's this release that's messed up

**Comment by ratwithacompiler** [comment]
Hi, sorry for the late reply, for some reason it didn't give me a notification before.

I've heard of a similar issue once from someone but couldn't reproduce it, could either of you upload a OBS log file from when you tried to stream with it maybe (OBS -> Help menu -> Log Files -> Show Log Files / View Current Logs) or otherwise give more details on what OBS version you're running (version num, 32/64bit,  etc). thanks!

**Comment by Swindyyy** [comment]
Here's my log from when I just ran up OBS now

https://obsproject.com/logs/MpbH__GvwDUahyhG

I'll check my others from when I'm streaming and get back to you in a moment

**Comment by MuzikxMonkey** [comment]
I just stepped out of my house but can help out in a little. It prolly didn't give a notification bc my account was flagged bc I had twitch handle in my profile. which I found odd. anyways. I'm on the latest OBS and I have tried all versions of this plug in. I hadn't tried going back to older OBS version due to other plugins I use. the best info I can share currently is my build is AMD I use an AMD encoder. I have tried recording and get no output file with subtitles. I hadn't had a chance but was curious if maybe it is the encoder causing the issue. but like I said, I will be back home shortly and can answer any questions from there. thank you again.

**Comment by Swindyyy** [comment]
Unfortunately I'm using the AMD encoder also. 

I'll try reinstalling this plug-in to see if it was just an error in my installation.

There's nothing I have to do on Twitch's side to say that I'll be using CC right?

**Comment by Swindyyy** [comment]
> Unfortunately I'm using the AMD encoder also.
> 
> I'll try reinstalling this plug-in to see if it was just an error in my installation.
> 
> There's nothing I have to do on Twitch's side to say that I'll be using CC right?

Nope, I just reinstalled the 0.0.11 and it doesn't work still. I feel like I'm missing something lol

**Comment by ratwithacompiler** [comment]
> Nope, I just reinstalled the 0.0.11 and it doesn't work still. I feel like I'm missing something lol

Nah nothing on Twitch required, if it shows captions in the dock or preview window and is set to output to streams in the settings it should just work.

But I'll try testing with hardware encoders, that could be the issue if you're both using the AMD encoder. 🤔

> have tried recording and get no output file with subtitles.
Playing subtitles in recordings can be a bit weird, even with VLC some builds don't show them and they don't work with many of the file formats, only ts files I think. That makes testing with recordings a bit inconvenient.

**Comment by MuzikxMonkey** [comment]
https://gyazo.com/49ee3b4e08c5fee5ee10f38d7774da1c
this is showing the dock working and capturing what i was saying
https://obsproject.com/logs/9AbRaWoLKhR4wD81
this is my log to showcase what might be going on.
I am back home currently and will be around for a while. I will note, i DO use a 2 pc set up, but all my audio, obs, everything is on the streaming pc and only i use NDI capture so, I'm not needing to route my audio any special way. I record in MKV format and remux to .mp4, neither of which showcase my subtitles. I have tried using x264 for encoding, no luck, I do use H264 for recording and streaming just to note tho. I have not personally tried streaming with to see if it would allow CC to work. Anything i may do to help you find a solution, i will be around tonight for a while, so feel free to fire questions if needed. Thank you again.

**Comment by MuzikxMonkey** [comment]
https://obsproject.com/logs/SZsjFH95YbU6_olH

just to give a cleaner log.

**Comment by ratwithacompiler** [comment]
@Swindyyy okay so @MuzikxMonkey tested with x264 now and that worked for him, while the AMD one didn't, so the issue is indeed the AMD encoder. 

Gonna try to figure out if it's possible to get it to work with that.
Thanks for testing @MuzikxMonkey!

**Comment by MuzikxMonkey** [comment]
All good. Anything to help.

**Comment by Swindyyy** [comment]
Oh it worked for him on x264? I read his earlier message to say that it
didn't work for him. I can give it a try on x264 tomorrow.

One thing I did notice, in the screenshots on the page showing it working,
the cricle is red but in my build, when I go live, the circle is grey
looking. Is that normal?

**Comment by MuzikxMonkey** [comment]
it was i tried x264 for recording, not streaming. I did a test stream using x264 and the CC worked. I swapped back to h264 and no luck.

**Comment by MuzikxMonkey** [comment]
As for the Red, mind doesnt show as Red either, i dont believe that is an issue, i think that is just part of Mac vs PC.

**Comment by willlllio** [comment]
Great so the person that I heard the issue first from a while ago also was using the GPU encoder and switching to x264 fixed it too.

And yeah the 🔴 is indeed just Win/Mac emoji looking different.

**Comment by ratwithacompiler** [comment]
Great so the person that I heard the issue first from a while ago also was using the GPU encoder and switching to x264 fixed it too.

And yeah the 🔴 is indeed just Win/Mac emoji looking different.

**Comment by MuzikxMonkey** [comment]
Hopefully it'll still be fixable.

**Comment by Swindyyy** [comment]
Any luck with this @ratwithacompiler?

**Comment by andmayfi92** [comment]
Hey guys just checking if this was ever resolved? I am trying on my older stream pc and it runs the AMD encoder.. pretty sure if I try to do x264 my old FX8350 might shit itself

**Comment by eduardofaria** [comment]
> Hey guys just checking if this was ever resolved? I am trying on my older stream pc and it runs the AMD encoder.. pretty sure if I try to do x264 my old FX8350 might shit itself

They deleted my issue about that, but no answer wroted about this... 😟
